src/Components/PaperEditor/TranscriptsContainer/TranscriptTabPane/TranscriptTabContent/index.js

```diff
@@ -424,38 +424,33 @@ const TranscriptTabContent = (props) => {
     });
   };
 
-  const onLabelCreate = (newLabel) => {
-    createLabel(newLabel);
-    const tempLabels = labels;
-    tempLabels.push(newLabel);
-    setLabels(tempLabels);
-    props.trackEvent({ category: 'paperEditor transcriptsTab', action: 'label create ' });
-  };
-
   const onLabelUpdate = (labelId, updatedLabel) => {
-    setLabels(() => [ ...tempLabels, updatedLabel ]);
+    setLabels(() => [ ...labels, updatedLabel ]);
     LabelsCollection.putItem(labelId, updatedLabel);
     props.trackEvent({ category: 'paperEditor transcriptsTab', action: `label update ${ labelId } ${ updatedLabel }` });
   };
 
-  const onLabelDelete = (labelId) => {
+  const onLabelDelete = (labelToDelete) => {
     const tempLabels = labels;
-    tempLabels.splice(labelId, 1);
+    if (activeLabel.id === labelToDelete.id) {
+      const defaultLabel = labels.find((label) => label.label === 'Default');
+      setActiveLabel(defaultLabel);
+    }
+    tempLabels.splice(labelToDelete.id, 1);
+    LabelsCollection.deleteItem(labelToDelete.id);
     setLabels(tempLabels);
-    LabelsCollection.deleteItem(labelId);
-    props.trackEvent({ category: 'paperEditor transcriptsTab', action: `label delete ${ labelId }` });
+    props.trackEvent({ category: 'paperEditor transcriptsTab', action: `label delete ${ labelToDelete }` });
```

Comment:
```suggestion
    trackEvent({ category: 'paperEditor transcriptsTab', action: `label delete ${ labelToDelete }` });
```
Do you need `props.` if you've destructured the props on line 16?